### ISSUE TITLE
TypeError: decode() argument 1 must be str, not bytes

### ISSUE THREAD

**Issue by ichorid** [Tribler/tribler#4957]
`tribler-7.4.0` branch, Linux, when clicking around the torrents list:

```python
ERROR   1574254475.49              rest_manager:83   (RESTRequest)  [Failure instance: Traceback: <class 'TypeError'>: decode() argument 1 must be str, not bytes
/home/vader/.pyenv/versions/tribler3/lib/python3.7/site-packages/twisted/web/http.py:2146:lineReceived
/home/vader/.pyenv/versions/tribler3/lib/python3.7/site-packages/twisted/web/http.py:2237:allContentReceived
/home/vader/.pyenv/versions/tribler3/lib/python3.7/site-packages/twisted/web/http.py:937:requestReceived
/home/vader/my_SRC/TRIBLER/tribler_ichorid/Tribler/Core/Modules/restapi/rest_manager.py:118:process
--- <exception caught here> ---
/home/vader/.pyenv/versions/tribler3/lib/python3.7/site-packages/twisted/web/server.py:217:process
/home/vader/.pyenv/versions/tribler3/lib/python3.7/site-packages/twisted/web/server.py:284:render
/home/vader/.pyenv/versions/tribler3/lib/python3.7/site-packages/twisted/web/resource.py:265:render
/home/vader/my_SRC/TRIBLER/tribler_ichorid/Tribler/Core/Modules/restapi/downloads_endpoint.py:294:render_GET
/home/vader/my_SRC/TRIBLER/tribler_ichorid/Tribler/Core/Modules/restapi/downloads_endpoint.py:111:get_files_info_json
/home/vader/my_SRC/TRIBLER/tribler_ichorid/Tribler/Core/DownloadState.py:167:get_files_completion
/home/vader/my_SRC/TRIBLER/tribler_ichorid/Tribler/Core/TorrentDef.py:422:get_files_with_length
/home/vader/my_SRC/TRIBLER/tribler_ichorid/Tribler/Core/TorrentDef.py:363:_get_all_files_as_unicode_with_length
/home/vader/my_SRC/TRIBLER/tribler_ichorid/Tribler/Core/TorrentDef.py:363:<listcomp>
/home/vader/my_SRC/TRIBLER/tribler_ichorid/Tribler/Core/Utilities/unicode.py:19:ensure_unicode
]
Traceback (most recent call last):
  File "/home/vader/.pyenv/versions/tribler3/lib/python3.7/site-packages/twisted/web/server.py", line 217, in process
    self.render(resrc)
  File "/home/vader/.pyenv/versions/tribler3/lib/python3.7/site-packages/twisted/web/server.py", line 284, in render
    body = resrc.render(self)
  File "/home/vader/.pyenv/versions/tribler3/lib/python3.7/site-packages/twisted/web/resource.py", line 265, in render
    return m(request)
  File "/home/vader/my_SRC/TRIBLER/tribler_ichorid/Tribler/Core/Modules/restapi/downloads_endpoint.py", line 294, in render_GET
    download_json["files"] = self.get_files_info_json(download)
  File "/home/vader/my_SRC/TRIBLER/tribler_ichorid/Tribler/Core/Modules/restapi/downloads_endpoint.py", line 111, in get_files_info_json
    files_completion = dict((name, progress) for name, progress in download.get_state().get_files_completion())
  File "/home/vader/my_SRC/TRIBLER/tribler_ichorid/Tribler/Core/DownloadState.py", line 167, in get_files_completion
    files = self.download.get_def().get_files_with_length()
  File "/home/vader/my_SRC/TRIBLER/tribler_ichorid/Tribler/Core/TorrentDef.py", line 422, in get_files_with_length
    for filename, length in self._get_all_files_as_unicode_with_length():
  File "/home/vader/my_SRC/TRIBLER/tribler_ichorid/Tribler/Core/TorrentDef.py", line 363, in _get_all_files_as_unicode_with_length
    yield (join(*[ensure_unicode(element, encoding) for element in file_dict[b"path"]]),
  File "/home/vader/my_SRC/TRIBLER/tribler_ichorid/Tribler/Core/TorrentDef.py", line 363, in <listcomp>
    yield (join(*[ensure_unicode(element, encoding) for element in file_dict[b"path"]]),
  File "/home/vader/my_SRC/TRIBLER/tribler_ichorid/Tribler/Core/Utilities/unicode.py", line 19, in ensure_unicode
    return s.decode(encoding, errors)
TypeError: decode() argument 1 must be str, not bytes
```

**Comment by xoriole** [comment]
Does it happen in Python3? I cannot reproduce on clicking torrents from discovered channels and search results?

**Comment by ichorid** [comment]
You should have some torrents with non-latin characters in filenames.

**Comment by ichorid** [comment]
I've tried to reproduce it from a clean slate (fresh .Tribler directory), and I can't. If I download something in Py3-7.4.0, no errors appear, but Cyrillic characters in filenames are shown with escape codes, like this:
![p3-7 4](https://user-images.githubusercontent.com/2509103/69245169-a0e41400-0ba6-11ea-8a37-00e3c61ce7e8.png)

So, I proceeded to try to reproduce it by downloading Cyrillic stuff with py2-7.3.2, and then upgrading to 7.4.0. **No torrents were converted from 7.3.2 to 7.4.0 at all!**

Thus, I'm closing this one, and open two new issues.
